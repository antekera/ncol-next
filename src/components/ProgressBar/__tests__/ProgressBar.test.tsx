import { render, screen } from '@testing-library/react'
import { ProgressBar } from '..'

jest.mock('@lib/context/StateContext', () => ({
  __esModule: true,
  default: () => ({ contentHeight: 1000 })
}))
jest.mock('@lib/hooks/useScrollProgress', () => ({
  useScrollProgress: () => 42
}))

describe('ProgressBar', () => {
  test('renders bar with width based on scroll progress', () => {
    render(<ProgressBar />)
    const bar = screen.getByTestId('progress-bar') as HTMLDivElement
    expect(bar.style.width).toBe('42%')
  })
})
