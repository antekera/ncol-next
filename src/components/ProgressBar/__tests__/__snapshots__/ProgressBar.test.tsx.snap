// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match snapshots 1`] = `
<div
  class="absolute left-0 w-full h-1 -bottom-[4px] bg-slate-200"
>
  <div
    class="h-full bg-primary ease-out duration-500"
    data-testid="progress-bar"
    style="width: 0%;"
  />
</div>
`;
